### PR TITLE
Combining segments to create new segment [MAILPOET-3469]

### PR DESCRIPTION
**Pull request by lysyjan**
[MAILPOET-3469]

[MAILPOET-3469]: https://mailpoet.atlassian.net/browse/MAILPOET-3469

*NOTES FOR QA*
- Please check that previously created segments can be opened and edited correctly with this new version.
- Design in the specification is before the redesign.

### CODE REVIEW

**Review by costasovo — CHANGES_REQUESTED** [review]
This is a huge amount of work! Good job! I added a couple of questions and potential issues.

File: tests/acceptance/Segments/ManageSegmentsCest.php [hunk 1]
```diff
@@ -105,12 +105,13 @@ public function createEditTrashRestoreAndDeleteExistingSegment(\AcceptanceTester
     $i->wantTo('Edit existing segment');
     $i->clickItemRowActionByItemName($segmentTitle, 'Edit');
     $i->waitForElementNotVisible('.mailpoet_form_loading');
-    $i->clearField('#field_name');
+    $i->clearField(['name' => 'name']);
+    $i->clearField(['name' => 'description']);
+    $i->wait(1);
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply tests/acceptance/Segments/ManageSegmentsCest.php:111]: Unfortunately, this wait is required. I will let a comment here.

File: tests/acceptance/Segments/ManageSegmentsCest.php [hunk 2]
```diff
@@ -135,6 +136,7 @@ public function createEditTrashRestoreAndDeleteExistingSegment(\AcceptanceTester
 
     $i->clickItemRowActionByItemName($segmentEditedTitle, 'Move to trash');
     $i->waitForText('1 segment was moved to the trash.');
+    $i->wait(1);
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply tests/acceptance/Segments/ManageSegmentsCest.php:142]: This wait makes the test more stable. I will let a comment here.

File: lib/Segments/DynamicSegments/Filters/UserRole.php [hunk 3]
```diff
@@ -9,6 +9,8 @@
 use MailPoetVendor\Doctrine\ORM\EntityManager;
 
 class UserRole implements Filter {
+  const TYPE = 'wordpressRole';
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply lib/Segments/DynamicSegments/Filters/UserRole.php:13]: This is not a type constant but an action constant. In DynamicSegmentFilterData are the type constants only. So I think that this filter is the right place for it.

File: assets/js/src/segments/dynamic/form.tsx [hunk 4]
```diff
@@ -92,31 +95,46 @@ export const Form: React.FunctionComponent<Props> = ({
...
+          ))}
           <Button
             type="button"
             dimension="small"
             variant="link"
             iconStart={plusIcon}
+            onClick={(e): void => {
+              e.preventDefault();
+              const filters = segment.filters;
+              filters.push({
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply assets/js/src/segments/dynamic/form.tsx:156]: It looks like this functionality is already present even before this PR; see current version at https://www.mailpoet.com/wp-admin/admin.php?page=mailpoet-segments#/new-segment. So I don't think it's necessary to fix it in this PR, but I'll make a note, and we might change it in the future.

File: lib/Entities/SegmentEntity.php [hunk 5]
```diff
@@ -150,4 +150,17 @@ public function getAverageEngagementScoreUpdatedAt(): ?\DateTimeInterface {
   public function setAverageEngagementScoreUpdatedAt(?\DateTimeInterface $averageEngagementScoreUpdatedAt): void {
     $this->averageEngagementScoreUpdatedAt = $averageEngagementScoreUpdatedAt;
   }
+
+  /**
+   * Returns connect operand from the first filter, when doesn't exist, then returns a default value.
+   * @return string
+   */
+  public function getFiltersConnectOperand(): string {
```
Comment (costasovo) [inline lib/Entities/SegmentEntity.php:158]: I think the correct name should be getFiltersConnectOperator.

**Review by costasovo — CHANGES_REQUESTED** [review]
Just one small detail.

**Review by costasovo — APPROVED** [review]
Great job! 👍

**Review by veljkho — APPROVED** [review]

**Pull request merged**